# HG changeset patch
# User Rik <rik@octave.org>
# Date 1371750945 25200
#      Thu Jun 20 10:55:45 2013 -0700
# Node ID 3d981b47be424ab81e9d0bfedd1ed279824423ea
# Parent  0fabe0f49f38f19e1407068f33869309cfb61cf3
Return button information for ginput when using FLTK (bug #32178)

* libinterp/dldfcn/__init_fltk__.cc(handle): Call execute_windowbuttondownfcn with
Fl::event_button() so that data is returned.

* scripts/plot/private/__fltk_ginput__.m: Pass button data back, rather than always
initializing to 1 (left mouse button).

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -1281,17 +1281,17 @@ private:
             gh = pixel2axes_or_ca (pos_x, pos_y);
 
             if (gh.ok ())
               {
                 ax_obj = gh_manager::get_object (gh);
                 set_axes_currentpoint (ax_obj, pos_x, pos_y);
               }
 
-            fp.execute_windowbuttondownfcn ();
+            fp.execute_windowbuttondownfcn (Fl::event_button()); 
 
             if (Fl::event_button () == 1 || Fl::event_button () == 3)
               return 1;
 
             break;
 
           case FL_DRAG:
             if (fp.get_windowbuttonmotionfcn ().is_defined ())
diff --git a/scripts/plot/private/__fltk_ginput__.m b/scripts/plot/private/__fltk_ginput__.m
--- a/scripts/plot/private/__fltk_ginput__.m
+++ b/scripts/plot/private/__fltk_ginput__.m
@@ -80,19 +80,17 @@ function [x, y, n, button] = ginput_aggr
   elseif (mode == 2)
     ## The end due to Enter.
     n = -1;
  endif
 endfunction
 
 function ginput_windowbuttondownfcn (src, data)
   point = get (get (src,"currentaxes"), "currentpoint");
-  ## FIXME -- How to get the actual mouse button pressed (1,2,3) into
-  ## "button"?
-  button = 1;
+  button = data;
   ginput_aggregator (1, point(1,1), point(2,1), button);
 endfunction
 
 function ginput_keypressfcn (src, evt)
   point = get (get (src, "currentaxes"), "currentpoint");
   ## FIXME -- use evt.Key or evt.Character?
   key = evt.Key;
   if (key == 10)
